samples/client/petstore/cpp-restsdk/client/model/Pet.h

```diff
@@ -31,6 +31,8 @@ namespace openapitools {
 namespace client {
 namespace model {
 
+class Category;
+class Tag;
```

Comment:
@dschmidt thanks for the PR. Are these changes expected?

Reply:
Thanks for coming back to me! 

Well, yes. In this specific case they are not strictly neccessary obviously, but they help in the general case. It's what this PR is all about: make it possible to create template which don't need model includes and avoid circular inclusion problems that way. 

I think having forward declarations that are not strictly needed  for everyone and every template is still much better than trying to be supersmart and to add complicated logic to determine whether they are needed or better than adding an option for it

Reply:
... and actually that should be the case for cpp-restsdk already  as it uses shared_ptr for storing data members. 

I. e.  in theory we should be able to move model includes to the cpp files already